mirrord/agent/src/steal/ip_tables/mesh.rs

```diff
@@ -114,53 +79,38 @@ where
     IPT: IPTables + Send + Sync,
 {
     async fn mount_entrypoint(&self) -> Result<()> {
-        self.preroute.mount_entrypoint().await?;
-
-        self.managed.inner().add_rule(
-            Self::ENTRYPOINT,
-            &format!("-j {}", self.managed.chain_name()),
+        tokio::try_join!(
+            self.prerouteing.mount_entrypoint(),
+            self.output.mount_entrypoint(),
         )?;
 
         Ok(())
     }
 
     async fn unmount_entrypoint(&self) -> Result<()> {
-        self.preroute.unmount_entrypoint().await?;
-
-        self.managed.inner().remove_rule(
-            Self::ENTRYPOINT,
-            &format!("-j {}", self.managed.chain_name()),
+        tokio::try_join!(
+            self.prerouteing.unmount_entrypoint(),
+            self.output.unmount_entrypoint(),
         )?;
 
         Ok(())
     }
 
     async fn add_redirect(&self, redirected_port: Port, target_port: Port) -> Result<()> {
-        self.preroute
-            .add_redirect(redirected_port, target_port)
-            .await?;
-
-        let redirect_rule = format!(
-            "{} -m tcp -p tcp --dport {redirected_port} -j REDIRECT --to-ports {target_port}",
-            self.own_packet_filter
-        );
-
-        self.managed.add_rule(&redirect_rule)?;
+        tokio::try_join!(
+            self.prerouteing.add_redirect(redirected_port, target_port),
+            self.output.add_redirect(redirected_port, target_port)
+        )?;
 
         Ok(())
     }
 
     async fn remove_redirect(&self, redirected_port: Port, target_port: Port) -> Result<()> {
-        self.preroute
-            .remove_redirect(redirected_port, target_port)
-            .await?;
-
-        let redirect_rule = format!(
-            "{} -m tcp -p tcp --dport {redirected_port} -j REDIRECT --to-ports {target_port}",
-            self.own_packet_filter
-        );
-
-        self.managed.remove_rule(&redirect_rule)?;
+        tokio::try_join!(
+            self.prerouteing
```

Comment:
do we want this to run concurrently? why?
should be quite quick and just makes it more complex per my understanding

Reply:
There is no real reason, It can be just sequential awaits

Reply:
I think it should be then, it'd be clearer